dipy/tracking/utils.py

```diff
@@ -937,3 +939,50 @@ def reduce_rois(rois, include):
             exclude_roi |= rois[i]
 
     return include_roi, exclude_roi
+
+
+def vals_from_img(img, streamlines, affine=None):
+    """
```

Comment:
OK - for now, I am moving it into tracking.streamline. I have also addressed the other comments -- I think that you are right about using the data array and an affine, rather than the nibabel image. Have refactored accordingly.